server/grammar/Yml.g4

```diff
@@ -152,7 +152,7 @@ ymlEntity:
  */
 expressionMarker: DOT DOT | DOT | MULTIVALUED_EXPRESSION;
 
-ymlId: YMLID | ARGS | LOCAL | RETURN;
+ymlId: YMLID | ARGS | LOCAL | RETURN | FUNCTION_AS_TYPE | TEXT_FUNCTION;
```

Comment:
Having `YMLID` and `ymlId` is confusing, btw. I’m having a hard time understanding what we’re dealing with, here.

Reply:
I'm dealing with the tokens `Function` and `TextFunction` that can be used as an introduction keyword for a function, or as the type of a function's argument.

Reply:
You misunderstood my comment. “What we’re dealing with, here” ⇔ “what this entity (ymlId) is”, not “what your change implies”. An ID that can be an argument list or a return or…? Wutwutwut

Reply:
The problem is that in YML most of the alpha-numeric expressions can be used as a valid identifier but at the same time YML has no reserved words. For example, I can have a local variable named `return`, `interface` or `function`, or even `Function`.
The only thing that makes sense for the YML compiler is the token order.
A specific token order has a specific meaning, but these tokens can still be used freely anywhere else.
That's the purpose of this ugly `ymlId` rule. It's more like a patch here. There is maybe a better solution, but I don't know it yet.

Reply:
But `ymlId` and `YMLID` seem very different from each other, hence the fact that I’m wondering if one of them could get a better name.

Reply:
YMLID (full caps) is a **lexer** rule. It explains how an id token is written.
ymlId is a **parser** rule. It accepts any YMLID and some other tokens that _could be_ YMLID but are more specialized tokens because I need them for some other **parser** rules.

Reply:
Yeah, yeah, but the simple fact that there’s an homonymy between a lexer rule and a parser rule bugs me xD

Reply:
Oh. I can either rename things right in this PR or do it in another, as you prefere.
The least impactful whould be to rename `YMLID` into `GENERIC_YMLID`.